runtime/str.go

```diff
@@ -221,13 +221,13 @@ func strFind(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	s := toStrUnsafe(args[0]).Value()
 	l := len(s)
 	start, end := 0, l
-	if argc >= 3 {
+	if argc >= 3 && args[2] != None {
```

Comment:
It occurred to me when I saw that CPython gives an error message about a slice index that they must be using some helper function related to slices. We have a similar helper, sliceClampIndex(), that handles None, checks for `__index__`, etc. I think we could just use that, something like:

```
if argc >= 3 {
  start, raised = sliceClampIndex(f, args[2], 0, l)
  if raised != nil {
    return nil, raised
  }
}
```